lib/ansiblelint/rules/PackageHasRetryRule.py

```diff
@@ -76,12 +76,31 @@ class PackageHasRetryRule(AnsibleLintRule):
         "absent",
     ]
 
+    _package_name_keys = [
+        "name",
+        "package",
+        "pkg",
+        "deb",
+    ]
+
     def matchtask(self, file, task):
         module = task["action"]["__ansible_module__"]
-        if module in self._package_modules:
-            if 'until' not in task:
-                # Exclude tasks where state is `absent`.
-                if task['action'].get('state') not in self._module_ignore_states:
-                    message = '{0} {1}'
-                    return message.format(self.shortdesc, module)
-                return False
+
+        if module not in self._package_modules or 'until' in task:
```

Comment:
I still insist on naming checks. Here I see just a list of some comparisons/checks. But it doesn't carry any human-understandable context.
Now it reads as "if some_var is missing from some struct or there's no until key in another struct".

If you name those checks with reasonable labels, it'd look different: "if module is a python package and the task is not retryable".
This makes more sense to the reader, isn't it?
Code is being read much more than being written. This is why we need to maintain readability and it's also one of the core coding principles when it comes to coding in Python.

```suggestion
        is_module_package = module in self._package_modules
        is_task_retryable = 'until' in task
        is_state_whitelisted = task['action'].get('state') in self._module_ignore_states
        
        is_retryable_whitelisted_package = (
            is_module_package
            and not is_task_retryable
            and is_state_whitelisted
        )
        if not is_retryable_whitelisted_package:
```